### PR TITLE
Update 0.8.3 acapy image and python versions

### PR DESCRIPTION
**Pull request by loneil**
This pull request:

- Updates to a 0.8.3 ACA-Py prerelease tag for Traction (will be included in the next official 0.9 release, discussed in meetings)
- Updates to the ACA-Py image that is askar-only (previous Traction code used the one that had Indy in it as well)
- Updates Python version to 3.9
- Some updates to the plugin builds for poetry

Changes are based on work @usingtechnology did in [PR 684](https://github.com/bcgov/traction/pull/684) but isolated and expanded on here to troubleshoot issues with the 0.8.2 askar-only image

Prior to 0.8.2 Traction was using `bcgovimages/aries-cloudagent:py36-1.16-1_0.8.1`
We are updating to use ghcr releases instead.
Initially tried upgrading to `ghcr.io/hyperledger/aries-cloudagent-python:py3.9-0.8.2` but this broke revocable credential creation. See discussions
https://github.com/hyperledger/aries-cloudagent-python/issues/2307
https://github.com/hyperledger/aries-cloudagent-python/pull/2308

We will use the prerelease image cut with the `indy_client` fix as discussed for the time being until 0.9
`ghcr.io/hyperledger/aries-cloudagent-python:py3.9-0.8.3-pre`

I've regression tested the environment here with this 0.8.3 release and it fixes the revocable creds (tails folder) problem and passes other functionality.

Issuer
f16d5181-5069-4c04-99b5-faac6dda9faa
38a720bd-1432-49d8-9557-541a6d4cab03

Non-Issuer
03490cb8-09e2-40f5-a4ef-0bd2bf6826c8
2fa1d27e-419e-46c5-8cd6-35aad9fdd0ce

### CODE REVIEW

**Comment by loneil** [comment]
Everything working at this time

- Reservation/Innkeeper approval/Tenant creation
- Make issuer (connect to endorser, register DID)
- Messaging
- Create schemas and cred defs (revocable and non)
- Receive proof requests

EXCEPT: Credentials (which offer all good, show up on the BC Wallet, or in the other tenant's credential exchange list), when accepted end up in abandoned

image

**Comment by loneil** [comment]
Also appears DID Exchange connection is sticking in `request`

image

**Comment by loneil** [comment]
Current point blocking issuing creds, tails file will not create

```
2023-07-17 21:07:23,889 aries_cloudagent.core.event_bus ERROR Error occurred while processing event
Traceback (most recent call last):
File "/home/aries/.venv/lib/python3.9/site-packages/aries_cloudagent/core/event_bus.py", line 121, in notify
await processor()
File "/home/aries/.venv/lib/python3.9/site-packages/aries_cloudagent/revocation/routes.py", line 1379, in on_revocation_registry_init_event
await shield(generate(registry_record))
File "/usr/local/lib/python3.9/asyncio/futures.py", line 284, in __await__
yield self # This tells Task to wait for completion.
File "/usr/local/lib/python3.9/asyncio/tasks.py", line 328, in __wakeup
future.result()
File "/usr/local/lib/python3.9/asyncio/futures.py", line 201, in result
raise self._exception
File "/usr/local/lib/python3.9/asyncio/tasks.py", line 256, in __step
result = coro.send(None)
File "/home/aries/.venv/lib/python3.9/site-packages/aries_cloudagent/revocation/routes.py", line 1323, in generate
await rr_record.generate_registry(profile)
File "/home/aries/.venv/lib/python3.9/site-packages/aries_cloudagent/revocation/models/issuer_rev_reg_record.py", line 185, in generate_registry
tails_hopper_dir = indy_client_dir(join("tails", ".hopper"), create=True)
File "/home/aries/.venv/lib/python3.9/site-packages/aries_cloudagent/indy/util.py", line 37, in indy_client_dir
makedirs(target_dir, exist_ok=True)
File "/usr/local/lib/python3.9/os.py", line 215, in makedirs
makedirs(head, exist_ok=exist_ok)
File "/usr/local/lib/python3.9/os.py", line 215, in makedirs
makedirs(head, exist_ok=exist_ok)
File "/usr/local/lib/python3.9/os.py", line 225, in makedirs
mkdir(name, mode)
PermissionError: [Errno 13] Permission denied: '/home/aries/.indy_client'
```

**Comment by loneil** [comment]
This will need the new 0.8.3 image with the `indy_client` folder fix

**Comment by swcurran** [comment]
Looks good to me, but I’m not at the right level to give the approval.

File: scripts/docker-compose.yml [hunk 1]
```diff
@@ -208,7 +208,7 @@ services:
       - host.docker.internal:host-gateway
 
   endorser-agent:
-    image: bcgovimages/aries-cloudagent:py36-1.16-1_0.8.1
+    image: ghcr.io/hyperledger/aries-cloudagent-python:py3.9-0.8.2
```
Comment (esune) [inline scripts/docker-compose.yml:211]: Might want to keep the image versions consistent between agent and endorser

**Review by esune — COMMENTED** [review]
Are we explicitly not supporting python 3.10/3.11 in `pyproject.toml`?

File: scripts/docker-compose.yml (hunk 1 above)
Comment (loneil) [reply scripts/docker-compose.yml:211]: Right, I had done that locally and never committed. Fixed for local docker. PRs and envs are still pointing to tools endorser, which is an older version as well (not even 0.8.2), but those should be pointed to other endorsers anyways soon.

**Comment by loneil** [comment]
> Are we explicitly not supporting python 3.10/3.11 in `pyproject.toml`?

I'd included that stuff over from Sherman's updates to the plugin python base images which switched from 3.6 to 3.9 images so I assume the goal is to keep everything in line to a certain version?

**Comment by usingtechnology** [comment]
i only limited it because the base images were at 3.9, seems safer to make that the same in the code. Wouldn't want someone writing code or bringing in a dependency that won't run on production images.

**Review by esune — APPROVED** [review]
👍🏻

**Pull request merged**